src/number/n_Q.jl

```diff
@@ -60,14 +60,8 @@ is_unit(n::n_Q) = !iszero(n)
 Return in `ZZ` the numerator of $x$.
 """
 function numerator(x::n_Q)
-   c = parent(x)
-   GC.@preserve x c QQ ZZ begin
-      xref = Ref(x.ptr)
-      p = libSingular.n_GetNumerator(xref, c.ptr)
-      x.ptr = xref[]
-      pp = libSingular.nApplyMapFunc(n_Q_2_n_Z, p, QQ.ptr, ZZ.ptr)
-      libSingular.n_Delete(p, QQ.ptr)
-      return ZZ(pp)
+   GC.@preserve x begin
+     return ZZ(libSingular.n_GetMPZ(x.ptr, parent(x).ptr))
```

Comment:
so.. is that always going to be a true MPZ, or are there optimizations in place for small values using tagged pointers that we need to take into account?

Reply:
in n_GetMPZ x.ptr should be of Singulars type number (which may be a "in place value" for small values),
but it returns always a true MPZ